src/system.rs

```diff
@@ -679,8 +680,24 @@ impl Resolver {
         let mut found_header = false;
         let mut found_footer = false;
 
+        // ISL version marker regex
+        let isl_version_marker = Regex::new(r"^\$ion_schema_\d+_\d+").unwrap();
```

Comment:
As per https://amzn.github.io/ion-schema/docs/isl-versioning#ion-schema-version-markers, the reserved space for ISL version markers is defined by this regex.
```suggestion
        let isl_version_marker = Regex::new(r"^\$ion_schema_\d.*$").unwrap();
```